### DISCUSSION TITLE
Can anyone provide an example using the TextRequestsWrapper node?

### DISCUSSION THREAD

**Discussion by TFWol** [langflow-ai/langflow#793] [Q&A]
I cannot figure out what it wants from me.
![image](https://github.com/logspace-ai/langflow/assets/9045213/2ea37225-688e-41e7-86d3-59e5af2357d6)

It looked straight forward at first.
At this point I'm just trying to make a simple POST and get a value back.

**Comment by TFWol** [comment]
Anyone? I want to like this project, but am I wasting my time?

**Reply by ogabrielluiz** [reply]
Hey, sorry for the delayed response.

This Requests wrapper is for passing headers to the tool. 
Since RequestsPostTool is a Tool, it is used by the agent to make requests.

**Reply by TFWol** [reply]
Thanks for the reply and sorry if I sounded exasperated; I'm trying to start relatively small with what I currently understand without falling down a rabbit hole.

I've been jumping between your project and flowise, but tend to come back to your project since I really like your approach and it looks more intuitive.
I just can't find many examples in general, especially on local API related tasks. 


What I'm specifically trying to do is use your project to interface with [LocalGPT's](https://github.com/PromtEngineer/localGPT) API (project uses LangChain)

For myself, it feels like a good starting point for interfacing with something local since it's just simple POST and GET requests and I don't have to spend too much time getting in the weeds; I'm trying to learn how to drive a car rather than build it.

The workflow I'm trying to understand is:
1. Type in a prompt (via chat or just predefined text in a block)
2. Have it passed via POST (I'm trying to send to an API)
3. Receive the response

Could you demonstrate what that would look like?


Btw, take a look at [NodeRed](https://github.com/node-red/node-red), a node-base home automation tool. I think it may be helpful if you need ideas for conceptual layouts and schemes for sharing flows in general.

**Reply by Sallery-X** [reply]
Could you provide an example? the examples is so poor~,

**Comment by Sallery-X** [comment]
> Since RequestsPostTool is a Tool, it is used by the agent to make requests. I want to using the TextRequestsWrapper  too

**Comment by carlosrcoelho** [comment]
Hey!

Thanks for sharing your question here. I'm sorry for not getting back to you sooner.
Langflow has gone through some major changes since then. I recommend trying out the new version for the best experience. If you have any issues, please open a new one; we'll be here to help.
